src/allmydata/watchdog/inotify.py

```diff
@@ -0,0 +1,139 @@
+        event_filepath_u = event.src_path.decode(encodingutil.get_filesystem_encoding())
+
+        if event_filepath_u == unicode_from_filepath(self._path):
+            # ignore events for parent directory
+            return
+        #try:
+        #    event_path = self._path.preauthChild(event.src_path)
+        #except InsecurePath, e:
+        #    print "failed: %r" % (e,)
+        #    return
+
+        def _maybe_notify(path):
+            if path in self._pending:
+                return
+            self._pending.add(path)
+            def _do_callbacks():
+                print "DO CALLBACKS"
+                self._pending.remove(path)
+                event_mask = IN_CHANGED
+                if event.is_directory:
+                    event_mask = event_mask | IN_ISDIR
+                for cb in self._callbacks:
+                    try:
+                        cb(None, FilePath(path), event_mask)
+                    except Exception, e:
+                        log.err(e)
+            _do_callbacks()
+        reactor.callFromThread(_maybe_notify, event_filepath_u)
+
+    def on_any_event(self, event):
+        print "PROCESS EVENT %r" % (event,)
+        self.process(event)
+
+class INotify(PollMixin):
+    """
+    I am a prototype INotify, made to work on Mac OS X (Darwin)
+    using the Watchdog python library. This is actually a simplified subset
+    of the twisted Linux INotify class because we do not utilize the watch mask
+    and only implement the following methods:
+     - watch
+     - startReading
+     - stopReading
+     - wait_until_stopped
+     - set_pending_delay
+    """
+    def __init__(self):
+        self._pending_delay = 1.0
+        self.recursive_includes_new_subdirectories = False
+        self._observer = Observer(timeout=self._pending_delay)
+        self._callbacks = {}
+        self._state = NOT_STARTED
+
+    def set_pending_delay(self, delay):
+        print "set pending delay"
+        self._pending_delay = delay
+        self._observer = Observer(timeout=self._pending_delay)
+
+    def startReading(self):
+        print "START READING BEGIN"
+        try:
+            _assert(len(self._callbacks) != 0, "no watch set")
+            self._observer.start()
+            self._state = STARTED
+        except Exception, e:
+            log.err(e)
+            self._state = STOPPED
+            raise
+        print "START READING END"
+
+    def stopReading(self):
+        print "stopReading begin"
+        if self._state != STOPPED:
+            self._state = STOPPING
+        self._observer.stop()
+        self._observer.join()
+        self._state = STOPPED
+        print "stopReading end"
+
+    def wait_until_stopped(self):
+        print "wait until stopped"
+        return self.poll(lambda: self._state == STOPPED)
+
+    def watch(self, path, mask=IN_WATCH_MASK, autoAdd=False, callbacks=None, recursive=False):
+        precondition(isinstance(autoAdd, bool), autoAdd=autoAdd)
```

Comment:
Since we're ignoring ``autoAdd`` (and not doing what it is supposed to do), assert that it is ``False``.